documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+
+*Default rebalance goals and cached proposal*
+
+The default rebalance goals are set in the `Kafka` custom resource. 
+You specify a list of goals in `spec.cruiseControl.config.default.goals`.
+
+Unless you set custom rebalance goals, the default rebalance goals are used by Cruise Control to return a _cached rebalance proposal_. 
+The cached proposal reflects the current state of the Kafka cluster and is updated every 15 minutes by default. 
+
+To review the list of default goals, see link:https://github.com/linkedin/cruise-control/wiki/Configurations#analyzer-configurations[Analyzer Configurations^] in the Cruise Control Wiki.
+
+For more information on setting the default rebalance goals, see xref:ref-cruise-control-configuration-{context}[]. 
+
+*Custom rebalance goals* 
+
+Custom rebalance goals are set in the `spec.goals` section of a `KafkaClusterRebalance` resource and override the default rebalance goals. 
+If no custom rebalance goals are set, the default rebalance goals are used.
+
+Use custom rebalance goals to generate a rebalance proposal that addresses a particular optimization scenario. 
+For example, you might want to optimize leader replica distribution across the Kafka cluster without considering other goals, such as disk or CPU usage. 
+So, you create a `KafkaClusterRebalance` resource containing a custom rebalance goal for leader replica distribution only.
+
+Using custom rebalance goals often returns a rebalance proposal faster than the default rebalance goals. 
+They must always be a subset of the default rebalance goals.
+
+For more information on setting custom rebalance goals, see xref:proc-generating-rebalance-proposals-{context}[].
+
+[discrete]
+== Rebalance proposals
+
+A _rebalance proposal_ is a summary of proposed changes from Cruise Control that would produce a more balanced Kafka cluster, with partition workloads distributed more evenly across the brokers. 
+Each rebalance proposal is based on the set of rebalance goals (either default goals or custom goals) that were used to generate it.
+
+Rebalance proposals always satisfy all the supplied hard goals, though not necessarily all the supplied soft goals.
+
+A rebalance proposal provides the following information in JSON:
+
+* An optimal mapping of partitions to brokers, generated by the Analyzer component of Cruise Control
+
+* Statistics on the cluster rebalance (before and after)
+
+You can use the information to decide whether to implement the cluster rebalance, or change the rebalance goals. 
+
+All rebalance proposals are dry runs. You cannot implement a cluster rebalance without first generating a rebalance proposal. 
+There is no limit to the number of rebalance proposals that can be generated.
+
+The following table lists the properties contained in a rebalance proposal:
+
+[cols="2*",options="header",stripes="none",separator=¦]
+|===
+
+m¦JSON property
+¦Description
```

Comment:
I will find some time to update the explainer docs with definitions of what the terms mean.